lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -148,7 +148,7 @@ def _parse_proxy_command(self, port=22):
             getattr(self._play_context, 'ssh_args', ''),
         ]
         if ssh_args is not None:
-            args = self._split_ssh_args(' '.join(ssh_args))
+            args = self._split_ssh_args(' '.join(filter(None,ssh_args)))
```

Comment:
I would argue that a None here means there is a bug on the caller, this would obscure that bug, not fix it.